Fix Quick start & Manual Docs (#5534)
STEllAR-GROUP/hpx: docs/sphinx/manual/building_hpx.rst

Software and libraries
----------------------

The table below presents all the necessary prerequisites for building |hpx|.

.. list-table:: Software prerequisites for |hpx|

   * * Name
     * Minimum version
   * * **Compilers**
     *
   * * |gcc|_
     * 7.0
   * * |clang|_
     * 7.0
   * * |visual_cxx|_ (x64)
     * 2015
   * * **Build System**
     *
   * * |cmake|_
     * 3.18
   * * **Required Libraries**
     *
   * * |boost|_
     * 1.71.0
   * * |hwloc|_
     * 1.5
   * * |asio|_
     * 1.12.0

The most important dependencies are |boost|_ and |hwloc|_. The installation of Boost 
is described in detail in Boost's `Getting Started <https://www.boost.org/more/getting_started/index.html>`_
document. A recent version of hwloc is required in order to support thread
pinning and NUMA awareness and can be found in |hwloc_downloads|_. 

|hpx| is written in 99.99% Standard C++ (the remaining 0.01% is platform
specific assembly code). As such, |hpx| is compilable with almost any standards

> compliant C++ compiler. The code base takes advantage of C++ language features when
> available (move semantics, rvalue references, magic statics, etc.). This may
> speed up the execution of your code significantly. We currently support the
> following C++ compilers: GCC, MSVC, ICPC and clang. For the status of your
> favorite compiler with |hpx| visit |hpx_buildbot|_.

Reviewer: The last few sentences could be removed:
```suggestion
compliant C++ compiler. The code base takes advantage of newer C++ language
and standard library features when available.
```

   localities. Please set the CMake variable ``MPI_CXX_COMPILER`` to your MPI
   C++ compiler wrapper if not detected automatically.

.. list-table:: Optional software prerequisites for |hpx|

   * * Name
     * Minimum version
   * * |google_perftools|_
     * 1.7.1
   * * |papi|

>      *

Reviewer: Could you move PAPI below gperftools, jemalloc, and mimalloc, please? gperftools, jemalloc, and mimalloc are all allocators so they can be grouped together.

Basic information
-----------------

The build system for |hpx| is based on |cmake|_, a cross-platform
build-generator tool which is not responsible for building the project 
but rather generates the files needed by your build tool (GNU make, Visual 
Studio, etc.) for building |hpx|. If CMake is not already installed in your 
system, you can download it and install it here: |cmake_download|_.

Once |cmake| has been run, the build process can be started. The |hpx| build
process is highly configurable through |cmake|, and various |cmake| variables
influence the build process. The build process consists of the following parts:

> * The |hpx| core libraries (target core): This forms the basic set of |hpx|

Reviewer 2: ```suggestion
* The |hpx| core libraries (target ``core``): This forms the basic set of |hpx|
```
(since the other targets have this font too)

   the ``CMAKE_BUILD_TYPE`` variables have to match and for projects not using
   |cmake|_, the ``HPX_DEBUG`` macro has to be set in debug mode.

.. _getting_hpx:

Getting |hpx|
-------------

Download a tarball of the latest release from |stellar_hpx_download|_ and
unpack it or clone the repository directly using ``git``:

.. code-block:: sh

    git clone https://github.com/STEllAR-GROUP/hpx.git

It is also recommended that you check out the latest stable tag:

.. code-block:: sh
    
    cd hpx

>     git checkout 1.7.1 

Reviewer 2: Agreed on the order. On the renaming I thought it was quite clear and straightforward as it is but I don't mind changing the names either

>     git checkout 1.7.1 

Author: So, I guess I leave the names as they are?

>     git checkout 1.7.1 

Reviewer: @dimitraka yeah, if we don't have better ideas at the moment let's leave them as they are.

              -DJEMALLOC_ROOT=/path/to/jemalloc
              [other CMake variable definitions]
              /path/to/source/tree

    For instance:

    .. code-block:: bash

        cmake -DBOOST_ROOT=~/packages/boost -DHWLOC_ROOT=/packages/hwloc -DCMAKE_INSTALL_PREFIX=~/packages/hpx ~/downloads/hpx_1.5.1

   * If you want to try |hpx| without using a custom allocator pass ``-DHPX_WITH_MALLOC=system`` to |cmake|:

    .. code-block:: sh 

        cmake -DCMAKE_INSTALL_PREFIX=/install/path -DHPX_WITH_MALLOC=system ..

    .. note::
       Please pay special attention to the section about :option:`HPX_WITH_MALLOC:STRING` as this is crucial for getting decent performance.

   .. important::

       If you are building |hpx| for a system with more than 64 processing units,
       you must change the |cmake| variable ``HPX_WITH_MAX_CPU_COUNT`` (to a value at least as big as the
       number of (virtual) cores on your system). Note that the default value is 64.

   .. caution::

       Compiling and linking |hpx| needs a considerable amount of memory. It is
       advisable that at least 2 GB of memory per parallel process is available.

   .. note::

>        Many Linux distributions use ``make`` as an alias for ``gmake``.

Reviewer 2: I think we agreed to remove this in the chat no? or maybe I don't recall correctly ^^
